### PR TITLE
Fixes issue #57

### PR DESCRIPTION
**Pull request by FlorianWilhelm**
This introduces a `SearchFilter` for filtering the search endpoint as described in issue #57.

### CODE REVIEW

File: src/notional/core.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ class ComposableObject(ModelMetaclass):
     Classes that support composition in this way must define and implement the internal
     `__compose__` method.  This method takes an arbitrary number of parameters, based
     on the needs of the implementation.  It is up to the implementing class to ensure
-    that the parameters are specificied correctly.
+    that the parameters are specified correctly.
```
Comment (jheddings) [inline src/notional/core.py:68]: Glad you are catching these!

**Review by jheddings — CHANGES_REQUESTED** [review]
This looks pretty straightforward...  Would you mind adding an example in `search.py`?

File: src/notional/query.py [hunk 2]
```diff
@@ -281,14 +289,19 @@ def filter(self, filter=None, **kwargs):
 
         if filter is None:
 
-            if "property" in kwargs:
-                filter = PropertyFilter.parse_obj(kwargs)
-            elif "timestamp" in kwargs and kwargs["timestamp"] == "created_time":
-                filter = CreatedTimeFilter.parse_obj(kwargs)
-            elif "timestamp" in kwargs and kwargs["timestamp"] == "last_edited_time":
-                filter = LastEditedTimeFilter.parse_obj(kwargs)
+            if isinstance(self.endpoint, SearchEndpoint):
```
Comment (jheddings) [inline src/notional/query.py:292]: Bummer that we need to look into the `notion_client` module to figure this out...  I can't think of an alternative at the moment.

File: src/notional/query.py [hunk 3]
```diff
@@ -281,14 +289,19 @@ def filter(self, filter=None, **kwargs):
...
+                filter = SearchFilter.parse_obj(kwargs)
             else:
-                raise ValueError("unrecognized filter")
+                if "property" in kwargs:
```
[the remark this answers is not in the document]
Comment (FlorianWilhelm) [reply src/notional/query.py]: Thanks, I don't know what made me do that 🤦 . Fixed it.

File: src/notional/query.py (hunk 2 above)
Comment (FlorianWilhelm) [reply src/notional/query.py:292]: Yeah, it's not so clean...

**Comment by FlorianWilhelm** [comment]
@jheddings, thanks for the review. I addressed your remark and extended the `search.py` example.

**Comment by jheddings** [comment]
Thanks for the bug fix!

**Pull request merged**